theseus_gui/src-tauri/src/api/profile_create.rs

```diff
@@ -0,0 +1,118 @@
+use crate::api::{Result, TheseusGuiError::ProfileCreation};
+use daedalus::modded::LoaderVersion;
+use futures::prelude::*;
+use std::path::PathBuf;
+use theseus::prelude::*;
+use tokio::fs;
+use tokio_stream::wrappers::ReadDirStream;
+
+const DEFAULT_NAME: &'static str = "Untitled Instance";
+const PROFILE_FILE_PATH: &'static str = "../.minecraft";
+
+// Generic basic profile creation tool.
+// Creates an essentially empty dummy profile with profile_create
+#[tauri::command]
+pub async fn profile_create_empty() -> Result<Profile> {
+    Ok(profile_create(
+        PathBuf::from(PROFILE_FILE_PATH), // the path of the newly created profile
+        String::from(DEFAULT_NAME),       // the name of the profile
+        String::from("1.8"),              // the game version of the profile
+        None,                             // the icon for the profile
+        ModLoader::Vanilla,               // the modloader to use
+        String::from("stable"), // the modloader version to use, set to "latest", "stable", or the ID of your chosen loader
+    )
+    .await?)
+}
+
+// Creates a profile at  the given filepath and adds it to the in-memory state
+// This is reused mostly from the CLI. TODO: touch up.
+// invoke('profile_add',profile)
+#[tauri::command]
+pub async fn profile_create(
+    path: PathBuf,          // the path of the newly created profile
+    name: String,           // the name of the profile
+    game_version: String,   // the game version of the profile
+    icon: Option<PathBuf>,  // the icon for the profile
+    modloader: ModLoader,   // the modloader to use
+    loader_version: String, // the modloader version to use, set to "latest", "stable", or the ID of your chosen loader
```

Comment:
I know this is separate in the CLI but because the code is shared this feels like it should be a library thing and just have a method called from Theseus